src/lib/components/Assessment/index.js

```diff
@@ -0,0 +1,232 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {html} from "lit-element";
+import {render} from "lit-html";
+import {BaseModalElement} from "../BaseModalElement";
+import "./_styles.scss";
+
+/**
+ * Element that renders a self-assessment callout.
+ * @extends {BaseModalElement}
+ * @final
+ */
+class Assessment extends BaseModalElement {
+  static get properties() {
+    return {
+      modal: {attribute: "aria-modal", reflect: true},
+    };
+  }
+
+  constructor() {
+    super();
+    this.modal = false;
+
+    this.renderLauncher = this.renderLauncher.bind(this);
+    this.onAssessmentAnimationEnd = this.onAssessmentAnimationEnd.bind(this);
+    this.onAssessmentResize = this.onAssessmentResize.bind(this);
+    this.reset = this.reset.bind(this);
+    this.onOpenClick = this.onOpenClick.bind(this);
+    this.openAssessment = this.openAssessment.bind(this);
+    this.closeAssessment = this.closeAssessment.bind(this);
+  }
+
+  render() {
+    if (!this.prerenderedChildren) {
+      this.prerenderedChildren = [];
+      this.setLeader = [];
+
+      for (const child of this.children) {
+        if (child.classList.contains("web-assessment__set-leader")) {
+          this.setLeader.push(child);
+        } else {
+          this.prerenderedChildren.push(child);
+        }
+      }
+    }
+
+    return html`
+      <div class="w-callout__header web-assessment__header">
+        <h2 class="w-callout__lockup web-assessment__lockup">
+          Check your understanding
+        </h2>
+        ${this.setLeader}
+        <button
+          @click="${this.onCloseClick}"
+          class="w-button--icon w-button--round web-assessment__close"
+          data-icon="close"
+        >
+          <span role="tooltip" class="w-tooltip">
+            Close
+          </span>
+        </button>
+      </div>
+      ${this.prerenderedChildren}
+    `;
+  }
+
+  firstUpdated() {
+    this.classList.remove("unresolved");
+    // Override BaseModalElement's inert behavior since Assessment
+    // is visible on desktop in closed state.
+    // (display: none used to remove it from the tab order when closed on mobile.)
+    this.inert = false;
+    // Render the launcher that appears in closed state on mobile.
+    this.renderLauncher();
+    // Listen to reset requests from child question components.
+    const questions = this.querySelectorAll("web-question");
+
+    for (const question of questions) {
+      question.addEventListener("request-assessment-reset", this.reset);
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    // Close modal if viewport is >481 px so document isn't inert on desktop
+    const mqString = `(min-width: 481px)`;
+
+    matchMedia(mqString).addListener(this.onAssessmentResize);
```

Comment:
I think you probably want to save the result of `matchMedia()` somewhere in the constructor (I think you're getting difference instances and the current `removeListener` does nothing), and add/remove listeners on connected/disconnected. And call `this.onAssesmentResize` when connected to establish initial state.

Reply:
I did everything but call `this.onAssessmentResize` when connected because (a) the assessment's initial state is established by `BaseModalElement` and (b) the assessment moves itself, so closing on connect means it can never open.